### PR TITLE
Change core worker C++ namespace to ray::core

### PR DESCRIPTION
**Pull request by raulchen**
## Why are these changes needed?

In order to avoid namespace conflicts between ray core code and C++ worker code, we wanted to change the namespace of all ray core code to "ray::core". However, this turned out to be too complex. 
Thus, this PR only changes core worker namespace, which should resolve most conflicts with the C++ worker. 

## Related issue number

#17388

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by qicosmos** [comment]
Also need to deal with such classes:
`
util/logging.h (RayLogLevel,RayLog)
cmmon/task/task.h(Task)
`

**Comment by raulchen** [comment]
> Also need to deal with such classes:
> 
> ```
> util/logging.h (RayLogLevel,RayLog)
> 
> cmmon/task/task.h(Task)
> ```

this PR only handles core worker code to avoid a huge PR

**Review by SongGuyang — APPROVED** [review]
LGTM.

**Review by scv119 — APPROVED** [review]

File: cpp/src/ray/config_internal.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
 namespace ray {
 namespace api {
 
+using ray::core::WorkerType;
```
Comment (kfstorm) [inline cpp/src/ray/config_internal.h:24]: Should we avoid `using` in header files?

File: src/ray/core_worker/core_worker.cc [hunk 2]
```diff
@@ -26,20 +26,20 @@
 #include "ray/util/util.h"
 
 namespace ray {
+namespace core {
 namespace {
```
Comment (kfstorm) [inline src/ray/core_worker/core_worker.cc:30]: Remove this line.

File: src/ray/core_worker/lib/java/jni_utils.h [hunk 3]
```diff
@@ -24,6 +24,9 @@
 #include "ray/common/ray_object.h"
 #include "ray/core_worker/core_worker.h"
 
+using namespace ray;
+using namespace ray::core;
```
Comment (kfstorm) [inline src/ray/core_worker/lib/java/jni_utils.h:28]: Nice change!

File: src/ray/core_worker/lib/java/jni_utils.h [hunk 4]
```diff
@@ -593,18 +594,18 @@ inline std::string GetFullName(bool global, std::string name) {
   if (name.empty()) {
     return "";
   }
-  return global ? name
-                : ::ray::CoreWorkerProcess::GetCoreWorker().GetCurrentJobId().Hex() +
-                      "-" + name;
+  return global
+             ? name
+             : ::CoreWorkerProcess::GetCoreWorker().GetCurrentJobId().Hex() + "-" + name;
```
Comment (kfstorm) [inline src/ray/core_worker/lib/java/jni_utils.h:599]:
```suggestion
             : CoreWorkerProcess::GetCoreWorker().GetCurrentJobId().Hex() + "-" + name;
```

**Review by kfstorm — COMMENTED** [review]
Could you list the results of `git grep ray:: src/ray/core_worker`? I'd like to know where the remaining usages of `ray::` prefix are in core worker code.

**Comment by kfstorm** [comment]
Since we already have `ray::gcs` and `ray::raylet`, do you plan to keep them or rename them to `ray::core::gcs` and `ray::core::raylet`? If you want to rename, should we move the core worker code to `ray::core::coreworker` in this PR?

File: cpp/src/ray/config_internal.h (hunk 1 above)
Comment (raulchen) [reply cpp/src/ray/config_internal.h:24]: This header is not exposed to users. So I think it's fine. we should avoid using in api headers.

**Comment by raulchen** [comment]
@kfstorm I tried changing everything, but that's too complex. I'd prefer keeping `ray::raylet` and `ray::gcs` for now.

**Comment by kfstorm** [comment]
@raulchen Do you plan to rename `ray::gcs` to `ray::core::gcs`in a later PR?

**Comment by raulchen** [comment]
we'll try changing other parts after releasing C++ worker.

**Pull request merged**